ykrt/src/mt.rs

```diff
@@ -429,31 +465,26 @@ impl MTThread {
 /// The innards of a meta-tracer thread.
 struct MTThreadInner {
     mt: MT,
-    /// A value that uniquely identifies a thread. Since this ID needs to be ORable with PHASE_TAG,
-    /// we use a pointer to a malloc'd chunk of memory. We guarantee a) that chunk is aligned to a
-    /// machine word b) that it is a non-zero chunk of memory (and thus guaranteed to be a unique
-    /// pointer).
-    tid: *mut u8,
+    /// An Arc whose pointer address uniquely identifies this thread. When a Location is traced,
+    /// this Arc's strong count will be incremented. If, when this thread drops, this Arc's strong
```

Comment:
I'm a bit confused by this bit. When this thread drops what? Itself?

Reply:
Would "when this thread is dropped" be clearer?

Reply:
Ah I see.

So when the `MTThreadInner` is dropped, we look at the strong count of the `Arc`. But won't it always be `>=1` at that point because the `MTThreadInner` has a handle on it? Either a typo, or I misunderstand.

Reply:
When a Location goes to `PHASE_TRACING` it increments the strong count. When this thread dies it decrements the strong count. When the Location goes from `PHASE_TRACING` to something else it decrement the strong count. Those two decrements can happen in any order.

Reply:
Yes, I get that bit.

Put differently, how do you inspect the strong count of an `Arc` and get a count of zero? Wouldn't that mean the `Arc` doesn't exist?

Reply:
And therefore the count has to be at least 1 if you are inspecting the strong count...

I bet I'm confused.

Reply:
The comment is quite careful not to claim that it inspecs the `Arc` to determine a strong count of zero. The strong count can only go to zero (but might not!) after the thread is dropped.

Reply:
Right. So, I wonder if this needs a reword, as it confused me.

Reply:
I think there's a limit to how clear this can be without becoming an essay, but https://github.com/softdevteam/yk/pull/187/commits/f07db556763e34c4d91e2a56c6871c590d4f2b27 might help a bit.

Reply:
You do yourself a disservice. That small change makes a world of difference!